src/sagemaker/local/image.py

```diff
@@ -1067,7 +1074,7 @@ def _ecr_login_if_needed(boto_session, image):
     ecr_url = auth["authorizationData"][0]["proxyEndpoint"]
 
     cmd = "docker login -u AWS -p %s %s" % (token, ecr_url)
-    subprocess.check_output(cmd.split())
+    subprocess.check_output(cmd.split(), stderr=subprocess.DEVNULL)
```

Comment:
Why do we set stderr to DEVNULL?

Reply:
[Suppressing the error](https://docs.python.org/3/library/subprocess.html#subprocess.check_call) if it errors. It will error if docker isn't configured properly ex. behind a proxy but don't have docker set up for it. 

The error has possibly sensitive creds in it. It was printing to build logs and triggering credential scanners. Also, might cause a notebook with the creds output to get commited.

I'm open to other ideas of how to suppress it.

Reply:
Gotcha,

That makes a lot of sense... I'm wondering if suppressing the error will cause issues for users to quickly debug/figure out why something failed.

Would it be possible to modify what is being sent to stderr? What are your thoughts?

Reply:
Good thought. 

Initially, I thought piping it might be a good idea but the docs recommend against using pipes with check_call. We could switch to using `Popen(cmd).communicate() ` with pipes, manually check the outputs and raise etc... 

It might be better to see if there's a different docker command to use for this? The only alternative I'm privy to is `["aws", "ecr", "get-login-password"]` (that's what I patch this out with currently). It relies on users having the ecr credentials helper, which I do not believe is configured on notebook instances. For studio, that might be even more of a pain to expect it to be in all images.

Maybe this belongs in a separate PR?

Reply:
Ahhh this solution looks like it requires a bit more time to figure out properly.

I agree, thank you so much for your efforts!

Can we remove `stderr=subprocess.DEVNULL` for now?

Reply:
Just pushed it. I'll open a separate issue for discussion.